models.js

```diff
@@ -482,103 +576,127 @@ debuglog = function () {
+        };
+    }
+
+    var binOpToXPath = printBinOp(objToXPath);
+    var binOpToHashtag = printBinOp(objToHashtag);
+
     var binOpChildren = function () {
         return [this.left, this.right];
     };
-    
+
     xpm.XPathBoolExpr = function(definition) {
         this.type = definition.type;
         this.left = definition.left;
         this.right = definition.right;
         this.toString = binOpToString;
         this.toXPath = binOpToXPath;
+        this.toHashtag = binOpToHashtag.bind(this);
         this.getChildren = binOpChildren;
         return this;
 
     };
-    
+
     xpm.XPathEqExpr = function(definition) {
         this.type = definition.type;
         this.left = definition.left;
         this.right = definition.right;
         this.toString = binOpToString;
         this.toXPath = binOpToXPath;
+        this.toHashtag = binOpToHashtag.bind(this);
         this.getChildren = binOpChildren;
         return this;
     };
-    
+
     xpm.XPathCmpExpr = function(definition) {
         this.type = definition.type;
         this.left = definition.left;
         this.right = definition.right;
         this.toString = binOpToString;
         this.toXPath = binOpToXPath;
+        this.toHashtag = binOpToHashtag.bind(this);
         this.getChildren = binOpChildren;
         return this;
     };
-    
+
     xpm.XPathArithExpr = function(definition) {
         this.type = definition.type;
         this.left = definition.left;
         this.right = definition.right;
         this.toString = binOpToString;
         this.toXPath = binOpToXPath;
+        this.toHashtag = binOpToHashtag.bind(this);
         this.getChildren = binOpChildren;
         return this;
     };
-    
+
     xpm.XPathUnionExpr = function(definition) {
         this.type = definition.type;
         this.left = definition.left;
         this.right = definition.right;
         this.toString = binOpToString;
         this.toXPath = binOpToXPath;
+        this.toHashtag = binOpToHashtag.bind(this);
         this.getChildren = binOpChildren;
         return this;
     };
-    
+
     xpm.XPathNumNegExpr = function(definition) {
         this.type = definition.type;
         this.value = definition.value;
         this.toString = function() {
             return "{unop-expr:" + this.type + "," + String(this.value) + "}";
         };
-        this.toXPath = function() {
-            return "-" + this.value.toXPath();
-        };
+        function _combine(transFunc) {
+            return function() {
+                return "-" + transFunc(this.value);
+            };
+        }
+        this.toXPath = _combine(objToXPath);
+        this.toHashtag = _combine(objToHashtag);
         this.getChildren = function () {
            return [this.value];
         };
         return this;
     };
-}());
+
+    return xpm;
+};
+
+var xpathmodels = makeXPathModels(defaultHashtagConfig);
+
+debuglog = function () {
+    if (xpathmodels.DEBUG_MODE) {
```

Comment:
Right now this references the global `xpathmodels`, but it should reference `xpm` in `makeXPathModels` (it should be moved inside `makeXPathModels`).